cmd/object-api-input-checks.go

```diff
@@ -0,0 +1,151 @@
+/*
+ * Minio Cloud Storage, (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"strings"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+)
+
+// Checks on GetObject arguments, bucket and object.
+func checkGetObjArgs(bucket, object string) error {
+	return checkBucketAndObject(bucket, object)
+}
+
+// Checks on DeleteObject arguments, bucket and object.
+func checkDelObjArgs(bucket, object string) error {
+	return checkBucketAndObject(bucket, object)
+}
+
+// Checks bucket and object validity, returns nil if both are valid.
+func checkBucketAndObject(bucket, object string) error {
+	// Verify if bucket is valid.
+	if !IsValidBucketName(bucket) {
+		return traceError(BucketNameInvalid{Bucket: bucket})
+	}
+	// Verify if object is valid.
+	if !IsValidObjectName(object) {
+		return traceError(ObjectNameInvalid{Bucket: bucket, Object: object})
+	}
+	return nil
+}
+
+// Checks for all ListObjects arguments validity.
+func checkListObjsArgs(bucket, prefix, marker, delimiter string, obj ObjectLayer) error {
+	// Verify bucket exists.
+	if err := checkBucketExist(bucket, obj); err != nil {
+		return traceError(err)
+	}
+	if !IsValidObjectPrefix(prefix) {
```

Comment:
Okay that change will happen here - https://github.com/minio/minio/pull/3381 - basically validating buckets right at the handler layer with in the lock rather than here at i/o layer.